src/cache.rs

```diff
@@ -21,6 +23,30 @@ pub struct Cache {
     os: OsType,
 }
 
+#[derive(Debug)]
+pub struct PageLookupResult {
+    page_path: PathBuf,
+    patch_path: Option<PathBuf>,
+}
+
+impl PageLookupResult {
+    fn with_page(page_path: PathBuf) -> Self {
+        Self {
+            page_path,
+            patch_path: None,
+        }
+    }
+
+    fn with_optional_patch(mut self, patch_path: Option<PathBuf>) -> Self {
+        self.patch_path = patch_path;
+        self
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = &Path> {
```

Comment:
Although I really love the idea to align this with common naming, I would rather call this something like `paths_to_print`.

Reply:
I agree about the more meaningful naming, but I'd stick with the simpler `paths`.